src/client/src/Client.c

```diff
@@ -146,6 +232,19 @@ STATUS createKinesisVideoClient(PDeviceInfo pDeviceInfo, PClientCallbacks pClien
 
     pKinesisVideoClient->base.pStateMachine = pStateMachine;
 
+    if(pKinesisVideoClient->deviceInfo.clientInfo.automaticStreamingFlags == AUTOMATIC_STREAMING_INTERMITTENT_PRODUCER) {
+        if ( !IS_VALID_TIMER_QUEUE_HANDLE(pKinesisVideoClient->timerQueueHandle) ) {
+            // Create timer queue
+            CHK_STATUS(timerQueueCreate(&pKinesisVideoClient->timerQueueHandle));
+            // Store callback in client so we can override in tests
+            pKinesisVideoClient->timerCallbackFunc = checkIntermittentProducerCallback;
+            CHK_STATUS(timerQueueAddTimer(pKinesisVideoClient->timerQueueHandle, 200 * HUNDREDS_OF_NANOS_IN_A_MILLISECOND,
+                                          pKinesisVideoClient->deviceInfo.clientInfo.callbackPeriod,

```

Comment:
Ah - I see why you are taking this in as you won't be able to reset within the test later. Here is my suggestion:

* Have a sentinel value defined as 0. 
* Add in public ClientInfo.callbackPeriod definition comments that the clients should set this field as 0. Perhaps, rename the field to reservedZero or reservedCallbackPeriod.
* In the constructor, check if it's 0 then use the defined value.